service/raft/raft_rpc.cc

```diff
@@ -154,11 +154,12 @@ raft_rpc::on_configuration_change(raft::server_address_set add, raft::server_add
         // Entries explicitly managed via `rpc::add_server` and
         // `rpc::remove_server` should never expire while entries
         // learnt upon receiving an rpc message should be expirable.
-        _address_map.set(addr.id, inet_addr, false);
+        _address_map.add_or_update_entry(addr.id, inet_addr);
+        _address_map.clear_expiring_flag(addr.id);
```

Comment:
Why split one function into two if those two are always used together?

Reply:
not always - we always add an expiring entry when get it from gossip, and only clear the expiring flag when handle raft notifications. When we load data from disk we need to add an entry as non-expiring right away. It's actually unclear if we should add it as a non-expiring one... Perhaps clear_expiring_flag is unnecessary.

Reply:
@gleb-cloudius

Reply:
I do not see what was wrong with an interface that has a flag. If we need to support all kinds of transitions it looks to be best. I do not have a strong feeling about it though.

Reply:
The whole API was renamed after a prompt from Kamil and Tomek to set_expiring/set_nonexpiring(). Resolving.